src/from-promise.ts

```diff
@@ -181,10 +181,9 @@ export function fromPromise<T>(
 
     promise.isPromiseBasedObservable = true
     promise.case = caseImpl
+    const oldState = oldPromise ? (oldPromise as any).state : undefined
     const oldData =
-        oldPromise && (oldPromise as any).state === FULFILLED
-            ? (oldPromise as any).value
-            : undefined
+        oldState === FULFILLED || oldState === PENDING ? (oldPromise as any).value : undefined
```

Comment:
So ... `oldState !== REJECTED` ? :smile:

Reply:
Yes, but I thought enumerating both would make the reason more explicit.

Reply:
Sure it doesn't really matter, feel free to keep it that way. I was just thinking about it the other way around ... if we carry over the previous promise value, why don't we carry over the errored value as well ... so from this perspective the REJECTED state acts as exception.